### PR TITLE
fix(web): Check circularity after mapping models

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Check circularity after mapping models

## What

* Data is allowed to be circular and the mapping should make sure to handle the circularity correctly
* This change checks for circularity after mapping the model instead of beforehand

## Why

* Some perfectly valid articles weren't getting indexed by the search-indexer since they contained entry-hyperlink fields that had circular references (the mapping would take care of that)

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/cms/src/lib/search/importers/article.service.ts [hunk 1]
```diff
@@ -111,11 +111,19 @@ export class ArticleSyncService implements CmsSyncProvider<IArticle> {
...
+          const mappedEntry = mapArticle(processedEntry)
+          if (!isCircular(mappedEntry)) {
+            processedEntries.push(processedEntry)
+          } else {
+            logger.warn('Circular reference found in article', {
+              id: entry?.sys?.id,
+            })
+          }
+        } catch (error) {
+          logger.warn('Failed to map article', {
```
Comment (siggifv) [inline libs/cms/src/lib/search/importers/article.service.ts:124]: Consider logging this as error. Same for the `Circular reference found in article` above.
Apparently, there are many places like this. We should discuss what we should consider as error vs. warning, keeping in mind that we want to know when some data is not being synced/updated, as there is highly likely that someone is making changes in Contentful and waiting for these changes to be appear on the web.

**Review by siggifv — CHANGES_REQUESTED** [review]

File: libs/cms/src/lib/search/importers/supportQNA.service.ts [hunk 2]
```diff
@@ -30,17 +30,19 @@ export class SupportQNASyncService implements CmsSyncProvider<ISupportQna> {
...
-            })
-          ) {
-            processedEntries.push(entry)
-          } else {
-            logger.warn('Circular reference found in question', {
-              id: entry.sys.id,
+          try {
+            const mappedEntry = mapSupportQNA(entry)
+            if (!isCircular(mappedEntry)) {
+              processedEntries.push(entry)
```
Comment (kristofer-lp8f) [inline libs/cms/src/lib/search/importers/supportQNA.service.ts:36]: Is this really supposed to be the non-mapped `entry`?
```suggestion
              processedEntries.push(mappedEntry)
```

File: libs/cms/src/lib/search/importers/news.service.ts [hunk 3]
```diff
@@ -13,13 +13,12 @@ export class NewsSyncService implements CmsSyncProvider<INews> {
   processSyncData(entries: processSyncDataInput<INews>) {
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply libs/cms/src/lib/search/importers/news.service.ts:13]: Right, this function could be called something else.

However for example in the article.service.ts file this function is actually processing something (it's removing fields) so one might argue that this function name might be right.

File: libs/cms/src/lib/search/importers/article.service.ts (hunk 1 above)
Comment (siggifv) [reply libs/cms/src/lib/search/importers/article.service.ts:124]: As per our discussions on Slack, we will not be logging these as errors, at least for now. Errors will call the operations team to action, which we don't want to do until we're confident that the indexer will not throw "too many" of those. In the meantime, we'll keep an eye on the logged warnings while we stabilize the indexer code.

**Review by siggifv — APPROVED** [review]

**Pull request merged**